impl/command_line/options.cxx

```diff
@@ -171,8 +197,65 @@ namespace substrate::commandLine
 		}, _option);
 	}
 
-	void option_t::displayHelp() const noexcept
-		{ console.writeln('\t', displayName(), ' ', _help); }
+	[[nodiscard]] size_t option_t::displayLength() const noexcept
+	{
+		return std::visit(match_t
+		{
+			[](const std::string_view &option) { return option.length(); },
+			[](const optionFlagPair_t &option)
+				{ return option._shortFlag.length() + option._longFlag.length() + 2U; },
+			[](const optionValue_t &option) { return option.metaName().length(); },
+		}, _option);
+	}
+
+	void option_t::displayHelp(const size_t padding) const noexcept
+	{
+		console.writeln('\t', displayName(), nullptr);
+		for ([[maybe_unused]] const auto _ : substrate::indexSequence_t{displayLength(), padding})
+			console.writeln(' ', nullptr);
+		console.writeln(' ', _help);
+	}
+
+	namespace internal
+	{
+		// This calculates how much padding is needed for these options to get a consistent padding
+		[[nodiscard]] size_t optionsHolder_t::displayPadding() const noexcept
+		{
+			return std::accumulate(begin(), end(), size_t{},
+				[](const size_t padding, const optionsItem_t &option) noexcept
+				{
+					return std::max(padding, std::visit(match_t
+						{
+							[](const option_t &value) { return value.displayLength(); },
+							[](const optionSet_t &value) { return value.displayPadding(); },
+						}, option)
+					);
+				}
+			);
+		}
+
+		void optionsHolder_t::displayHelp() const noexcept
+		{
+			// Figure out how much padding is needed to make everything neat
+			const auto padding{displayPadding()};
+			std::vector<optionSet_t> optionSets{};
+
+			// Now display the non-alternation options and collect option sets
+			console.writeln("Options:"sv);
+			for (const auto &option : *this)
+			{
+				std::visit(match_t
+				{
+					[=](const option_t &value) { value.displayHelp(padding); },
+					[&](const optionSet_t &value) { optionSets.emplace_back(value); },
```

Comment:
Nit: emplace_back always triggers a reallocation check (since you don't set a max capacity). Apart from that, see: https://groups.google.com/a/chromium.org/g/chromium-dev/c/7mJypsYz6AA